Install script breaks Github action in 2.61

### Describe the bug

We are running the install script at `https://aka.ms/InstallAzureCLIDeb` in a github action pipeline like this:
```
      - name: Install Azure CLI
        run: |
          sudo apt update && sudo apt-get install curl -y
          curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
```

this morning this was broken, and it seems like the reason is because the gpg installation breaks the script with
```
gpg: cannot open '/dev/tty': No such device or address
curl: (23) Failed writing body
Error: Process completed with exit code 2.
```

I did some digging by enabling a tty and it seems the gpg installation wants to ask for an okay to overwrite the file:
`File '/etc/apt/keyrings/microsoft.gpg' exists. Overwrite? (y/N)`

which I assume is due to the path changes in:
`https://github.com/Azure/azure-cli/commit/655b05b15204ef8a5c76b7a5cd78cb489058530a`

### Related command

https://aka.ms/InstallAzureCLIDeb

### Errors

gpg: cannot open '/dev/tty': No such device or address
curl: (23) Failed writing body
Error: Process completed with exit code 2.

### Issue script & Debug output

not possible

### Expected behavior

the install script installs az cli without error

### Environment Summary

github action standard runner, before installation

### Additional context

_No response_

--- a/scripts/release/debian/deb_install.sh
+++ b/scripts/release/debian/deb_install.sh
@@ -49,7 +49,7 @@ setup() {
     set -v
     mkdir -p /etc/apt/keyrings
     curl -sLS https://packages.microsoft.com/keys/microsoft.asc |
-      gpg --dearmor -o /etc/apt/keyrings/microsoft.gpg
+      gpg --dearmor > /etc/apt/keyrings/microsoft.gpg
     chmod go+r /etc/apt/keyrings/microsoft.gpg
     set +v
 

